lib/slide_hub/batch.rb

```diff
@@ -24,68 +25,105 @@ def self.execute
+        true
       else
         false
       end
-      SlideHub::BatchLogger.info('Start converting from PPM to JPG')
-      slide_image_list = SlideHub::ConvertUtil.new.ppm_to_jpg(dir)
-      SlideHub::BatchLogger.info(SlideHub::ConvertUtil.new.get_local_file_list(dir, '').inspect)
-      final_list = slide_image_list.dup
+    end
 
-      self.generate_json_list(slide_image_list, key, "#{dir}/list.json")
-      final_list.push("#{dir}/list.json")
+    def convert_to_jpg
+      SlideHub::BatchLogger.info('Start converting from PPM to JPG')
+      @slide_image_list = SlideHub::ConvertUtil.new.ppm_to_jpg(@work_dir)
+      SlideHub::BatchLogger.info(SlideHub::ConvertUtil.new.get_local_file_list(@work_dir, '').inspect)
+      @upload_file_list = @slide_image_list.dup
+    end
 
+    def convert_to_thumbnail
       SlideHub::BatchLogger.info('Generating thumbnails')
-      SlideHub::BatchLogger.info(slide_image_list.inspect)
-      thumbnail_list = SlideHub::ConvertUtil.new.jpg_to_thumbnail(slide_image_list)
+      SlideHub::BatchLogger.info(@slide_image_list.inspect)
+      thumbnail_list = SlideHub::ConvertUtil.new.jpg_to_thumbnail(@slide_image_list)
       thumbnail_list.each do |tm|
-        final_list.push(tm)
+        @upload_file_list.push(tm)
       end if thumbnail_list.instance_of?(Array)
+    end
 
-      transcript = SlideHub::ConvertUtil.new.pdf_to_transcript(dir, "#{file}.pdf")
-      final_list.push(transcript) if transcript
-
-      SlideHub::BatchLogger.info(final_list.inspect)
-      CloudConfig::SERVICE.upload_files(CloudConfig::SERVICE.config.image_bucket_name, final_list, key)
+    def convert_to_transcript
+      transcript = SlideHub::ConvertUtil.new.pdf_to_transcript(@work_dir, "#{@work_file}.pdf")
+      @upload_file_list.push(transcript) if transcript
+    end
 
-      slide = Slide.where('slides.key = ?', key).first
-      slide.convert_status = 100
-      slide.extension = ".#{ft}"
-      slide.num_of_pages = slide_image_list.count
-      slide.save
-      true
+    def generate_json
+      save_list = []
+      @slide_image_list.each do |item|
+        save_list.push("#{@key}/#{File.basename(item)}")
+      end
+      open("#{@work_dir}/list.json", 'w') do |io|
+        JSON.dump(save_list, io)
+      end
+      @upload_file_list.push("#{@work_dir}/list.json")
     end
-  end
 
-  def self.generate_json_list(list, prefix, filename)
-    save_list = []
-    list.each do |item|
-      save_list.push("#{prefix}/#{File.basename(item)}")
+    def upload_files
+      SlideHub::BatchLogger.info(@upload_file_list.inspect)
+      CloudConfig::SERVICE.upload_files(CloudConfig::SERVICE.config.image_bucket_name, @upload_file_list, @key)
     end
-    open(filename, 'w') do |io|
-      JSON.dump(save_list, io)
+
+    def update_database
+      slide = Slide.where('slides.key = ?', @key).first
+      slide.convert_status = 100
+      slide.extension = ".#{@file_type}"
+      slide.num_of_pages = @slide_image_list.count
```

Comment:
__[reek]__
* [FeatureEnvy] refers to slide more than self (maybe move it to another class?) [:ref](https://github.com/troessner/reek/blob/master/docs/Feature-Envy.md)

<!-- Comment by [SideCI](https://sideci.com) -->